junit-jupiter-migration-support/src/main/java/org/junit/jupiter/migrationsupport/rules/ParameterizedExtension.java

```diff
@@ -0,0 +1,265 @@
+	 * this extension once again here only support them on the constructor and require an {@code @Parameters} method
+	 *
+	 * @param parameterContext the context for the parameter to be resolved; never
+	 * {@code null}
+	 * @param extensionContext the extension context for the {@code Executable}
+	 * about to be invoked; never {@code null}
+	 * @return true if the above is met otherwise false.
+	 */
+	@Override
+	public boolean supports(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		return hasParametersMethod(extensionContext)
+				&& parameterContext.getDeclaringExecutable() instanceof Constructor;
+	}
+
+	@Override
+	public Object resolve(ParameterContext parameterContext, ExtensionContext extensionContext)
+			throws ParameterResolutionException {
+		int parameterCount = parameterContext.getDeclaringExecutable().getParameterCount();
+		Object[] parameters = resolveParametersForConstructor(extensionContext, parameterCount);
+
+		int parameterIndex = parameterContext.getIndex();
+		//move to the next set of parametersFields
+		if (lastParameterToBeResolved(parameterContext)) {
+			this.parametersCollectionIndex++;
+		}
+
+		return parameters[parameterIndex];
+	}
+
+	/**
+	 * Retrieves the Object[] of the current iteration we are working on.
+	 *
+	 * @param extensionContext the extensionContext
+	 * @param parameterCount the amount of parameters of the constructor.
+	 *
+	 * @return the object[] for this parameter iteration.
+	 * @throws ParameterResolutionException If the amount of arguments of the constructor doesn't match the amount
+	 * of arguments of the currently resolved object[]
+	 */
+	private Object[] resolveParametersForConstructor(ExtensionContext extensionContext, int parameterCount)
+			throws ParameterResolutionException {
+		return parameters(extensionContext).map(ArrayList::new).map(l -> l.get(this.parametersCollectionIndex)).filter(
+			params -> params.length == parameterCount).orElseThrow(
+				ParameterizedExtension::unMatchedAmountOfParametersException);
+	}
+
+	private static boolean hasCorrectParameterFields(ExtensionContext context) {
+		List<Field> fields = parametersFields(context);
+		boolean hasFieldInjection = !fields.isEmpty();
+
+		if (hasArgsConstructor(context) && hasFieldInjection) {
+			return false;
+		}
+		else if (hasFieldInjection) {
+			return areParametersFormedCorrectly(fields);
+		}
+
+		return true;
+	}
+
+	private static boolean areParametersFormedCorrectly(List<Field> fields) {
+		List<Integer> parameterValues = parameterIndexes(fields);
+
+		List<Integer> duplicateIndexes = duplicatedIndexes(parameterValues);
+
+		boolean hasAllIndexes = indexRangeComplete(parameterValues);
+
+		return hasAllIndexes && duplicateIndexes.isEmpty();
+	}
+
+	private static List<Integer> parameterIndexes(List<Field> fields) {
+		return fields.stream().map(f -> f.getAnnotation(Parameterized.Parameter.class)).map(
+			Parameterized.Parameter::value).collect(toList());
```

Comment:
Consider using `// @formatter:{off|on}` here and all the other stream-using methods below.